pkg/controllers/secretstore/clustersecretstore_controller.go

```diff
@@ -71,12 +86,178 @@ func (r *ClusterStoreReconciler) Reconcile(ctx context.Context, req ctrl.Request
+func (r *ClusterStoreReconciler) hasPushSecretsWithDeletePolicy(ctx context.Context, css *esapi.ClusterSecretStore) (bool, error) {
+	storeName := css.GetName()
+
+	// Use the index to find PushSecrets that have pushed to this store
+	var pushSecretList esv1alpha1.PushSecretList
+	if err := r.List(ctx, &pushSecretList, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("status.syncedPushSecrets", storeName),
+		// No namespace filter for ClusterSecretStore - look in all namespaces
+	}); err != nil {
+		return false, fmt.Errorf("failed to list PushSecrets by store index: %w", err)
+	}
+
+	// Check if any of these PushSecrets have DeletionPolicy=Delete
+	for _, ps := range pushSecretList.Items {
+		if ps.Spec.DeletionPolicy == esv1alpha1.PushSecretDeletionPolicyDelete {
+			// Verify the store reference matches
+			storeKey := fmt.Sprintf("%s/%s", esapi.ClusterSecretStoreKind, storeName)
+			if _, hasPushed := ps.Status.SyncedPushSecrets[storeKey]; hasPushed {
+				return true, nil
+			}
+		}
+	}
+
+	// Also check for PushSecrets that reference this store by name or labelSelector
+	// but haven't pushed to it yet (for initial finalizer setup)
+	if err := r.List(ctx, &pushSecretList, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("spec.deletionPolicy", string(esv1alpha1.PushSecretDeletionPolicyDelete)),
+		// No namespace filter for ClusterSecretStore - look in all namespaces
+	}); err != nil {
+		return false, fmt.Errorf("failed to list PushSecrets by deletionPolicy: %w", err)
+	}
+
+	for _, ps := range pushSecretList.Items {
+		// Check if this PushSecret references our store
+		for _, storeRef := range ps.Spec.SecretStoreRefs {
+			if storeRef.Name == storeName && storeRef.Kind == esapi.ClusterSecretStoreKind {
+				return true, nil
+			}
+		}
+
+		// Check labelSelector match
+		for _, storeRef := range ps.Spec.SecretStoreRefs {
+			if storeRef.LabelSelector != nil && storeRef.Kind == esapi.ClusterSecretStoreKind {
+				selector, err := metav1.LabelSelectorAsSelector(storeRef.LabelSelector)
+				if err != nil {
+					continue
+				}
+				if selector.Matches(labels.Set(css.GetLabels())) {
+					return true, nil
+				}
+			}
+		}
+	}
+
+	return false, nil
+}
+
 // SetupWithManager returns a new controller builder that will be started by the provided Manager.
 func (r *ClusterStoreReconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("cluster-secret-store")
 
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(opts).
 		For(&esapi.ClusterSecretStore{}).
+		Watches(
+			&esv1alpha1.PushSecret{},
+			handler.EnqueueRequestsFromMapFunc(r.findClusterSecretStoresForPushSecret),
+			builder.WithPredicates(predicate.NewPredicateFuncs(func(obj client.Object) bool {
+				// Only watch PushSecrets with DeletionPolicy=Delete
+				ps := obj.(*esv1alpha1.PushSecret)
+				return ps.Spec.DeletionPolicy == esv1alpha1.PushSecretDeletionPolicyDelete
```

Comment:
nit: this predicate is a no-op because predicates are client-side only and you already check for this in your enqueue function https://github.com/external-secrets/external-secrets/pull/5163/files#diff-28773406fab3a6c16378745a672b49a6cf756f75db8f794c9ebb5886a44dd521R84-R86